src/main/scala/co/teapot/tempest/typedgraph/SimpleTypedGraph.scala

```diff
@@ -0,0 +1,29 @@
+package co.teapot.tempest.typedgraph
+
+import co.teapot.tempest.Node
+import co.teapot.tempest.graph.DirectedGraph
+
+/** A SimpleTypedGraph has a single source node type and single target node type. */
+case class SimpleTypedGraph(sourceNodeType: String, targetNodeType: String, graph: DirectedGraph) extends TypedGraph {
+  override def outNeighbors(node: Node): Traversable[Node] = new Traversable[Node] {
+    def foreach[A](f: Node => A): Unit = {
+      // In use cases like the union graph, we might give this simple graph a node whose type doesn't match, in which
+      // case outNeighbors should do nothing.
```

Comment:
Right, the empty list (or technically the empty Traversable, which does nothing when you call foreach on it).  I think that's the right thing to do.